src/main/groovy/org/jenkinsci/gradle/plugins/jpi/LicenseTask.groovy

```diff
@@ -0,0 +1,94 @@
+package org.jenkinsci.gradle.plugins.jpi
+
+import groovy.xml.MarkupBuilder
+import groovy.xml.QName
+import org.gradle.api.DefaultTask
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.artifacts.ModuleVersionIdentifier
+import org.gradle.api.artifacts.ResolvedArtifact
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+import org.gradle.internal.component.local.model.DefaultProjectComponentIdentifier
+
+class LicenseTask extends DefaultTask {
+    static final String TASK_NAME = 'generateLicenseInfo'
+
+    private Set<Configuration> configurations
+
+    Set<Configuration> providedConfigurations
+
+    @OutputDirectory
+    File outputDirectory
+
+    void setConfigurations(Set<Configuration> configurations) {
+        this.configurations = configurations
+        this.inputs.files(configurations*.incoming.files)
```

Comment:
Why don't you add a getter returning `configurations*.incoming.files` which is annotated with `@InputFiles`?
I think this code causes resolving the configuration whenever it is set at configuration time.